fog/test-client/src/test_client.rs

```diff
@@ -265,128 +330,412 @@ impl TestClient {
+                            }
+                            if memo.get_fee() != fee {
+                                log::error!(
+                                    self.logger,
+                                    "Destination memo had wrong fee, found {}, expected {}",
+                                    memo.get_fee(),
+                                    fee
+                                );
+                                return Err(TestClientError::UnexpectedMemo);
+                            }
+                            if memo.get_num_recipients() != 1 {
+                                log::error!(self.logger, "Destination memo had wrong num_recipients, found {}, expected 1", memo.get_num_recipients());
+                                return Err(TestClientError::UnexpectedMemo);
+                            }
+                            if *memo.get_address_hash() != tgt_address_hash {
+                                log::error!(self.logger, "Destination memo had wrong address hash, found {:?}, expected {:?}", memo.get_address_hash(), tgt_address_hash);
+                                return Err(TestClientError::UnexpectedMemo);
+                            }
+                        }
+                        _ => {
+                            log::error!(self.logger, "Source Client: Unexpected memo type");
+                            return Err(TestClientError::UnexpectedMemo);
+                        }
                     }
-                },
-                Ok(None) => {
-                    panic!("source client didn't find destination memo");
                 }
-                Err(err) => {
-                    panic!("source client memo error: {}", err);
-                }
-            }
-
-            // Ensure target client got a sender memo, as expected for recoverable
-            // transcation history
-            match target_client.get_last_memo() {
-                Ok(Some(memo)) => match memo {
-                    MemoType::AuthenticatedSender(memo) => {
-                        assert_eq!(
-                            memo.sender_address_hash(),
-                            ShortAddressHash::from(
-                                &source_client.get_account_key().default_subaddress()
-                            )
-                        );
-                    }
-                    _ => {
-                        panic!("unexpected memo type")
-                    }
-                },
                 Ok(None) => {
-                    panic!("target client didn't find sender memo");
+                    log::error!(self.logger, "Source Client: Missing memo");
+                    return Err(TestClientError::UnexpectedMemo);
                 }
                 Err(err) => {
-                    panic!("target client memo error: {}", err);
+                    log::error!(self.logger, "Source Client: Memo error: {}", err);
```

Comment:
Would it be misleading if we made this message also: log::error!(logger, "Source Client: Memo parse error: {}", err);
This seems like it'd be more consistent with what we do for the error message we give for Target Client errors.